dsgrid/project.py

```diff
@@ -155,10 +159,29 @@ def unload_dataset(self, dataset_id):
         """
         self._datasets.pop(dataset_id, None)
 
-    def transform_dataset(self, dataset_id: str) -> DataFrame:
-        """Transform a dataset by mapping its dimensions to the project's dimensions."""
+    def transform_dataset(
+        self,
+        dataset_id: str,
+        output_dir: Path,
+        overwrite=False,
+        scratch_dir: Optional[Path] = None,
+    ) -> DataFrame:
+        """Transform a dataset by mapping its dimensions to the project's dimensions and write it
+        to the filesystem.
+        """
+        output_file = output_dir / "table.parquet"
```

Comment:
This is an unused function. It provides the ability to map a dataset to project dimensions without a query. It could be supplanted with an auto-generated query. Anyway, in order to work, we have to persist the dataframe to files in a permanent location because we may be relying on temporary files (the scratch directory gets deleted when the function exits.

I made a mistake in the return. We should either return the path alone or the dataframe and the path.